save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -233,85 +249,35 @@ class WarnPlugin(
      */
     private fun Path.collectWarningsWithLineNumbers(
         warnPluginConfig: WarnPluginConfig,
-        generalConfig: GeneralConfig
+        generalConfig: GeneralConfig,
+        testFiles: List<Path>,
     ): List<Warning> {
-        val linesFile = fs.readLines(this)
-        return generalConfig.expectedWarningsEndPattern?.let {
+        return if (warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF) {
+            val sarifFileName = warnPluginConfig.expectedWarningsFileName!!
+            val sarif = fs.findSarifUpper(this, sarifFileName)
+                ?: error(
+                    "Could not find SARIF file with expected warnings for file $this. " +
+                            "Please check if correct `expectedWarningsFormat` is set and if the file is present and called `$sarifFileName`."
+                )
+            val topmostTestDirectory = fs.topmostTestDirectory(this)
```

Comment:
We didn't need test root in any form inside plugins before, because plugins used to work only with their own resources. However now, since we introduce an external file, that can and will contain results for many tests, we need to have this information here. It should probably be done in `Plugin` class itself